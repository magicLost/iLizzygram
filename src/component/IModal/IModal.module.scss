.modal {
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: auto;
  border: none;
  background-color: transparent;
}
.wrapperForm {
  position: relative;
  outline: 0;
  &:focus {
    outline: 0 !important ;
  }
  overflow: auto;
  max-height: 80%;
  background-color: white;
  padding: 50px 25px 25px;
}

.wrapperSlider {
  position: relative;
  outline: 0;
  &:focus {
    outline: 0 !important ;
  }
  overflow: auto;
  //maxHeight:  80% ;
  background-color: white;
  //padding:  50px 25px 25px ;
  width: 100%;
}
