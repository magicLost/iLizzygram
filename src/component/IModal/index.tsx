import React from "react";
import classes from "./IModal.module.scss";
//import { makeStyles } from "@material-ui/core/styles";
//import CircularProgress from "@material-ui/core/CircularProgress";
//import Button from "@material-ui/core/Button";
//import classes from './LoadableBackDrop.module.scss';
import Modal from "@material-ui/core/Modal";
import ModalCloseButton from "../UI/ModalCloseButton";

type TModalType = "slider" | "form";

interface IModalProps {
  open: boolean;
  type: TModalType;
  onClose: () => void | undefined;
  children: any;
}
/* 
const useStyles = makeStyles({
  modal: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    overflow: "auto",
    border: "none",
    backgroundColor: "transparent",
  },
  wrapperForm: {
    position: "relative",
    outline: 0,
    "&:focus": {
      outline: "0 !important",
    },
    overflow: "auto",
    maxHeight: "80%",
    backgroundColor: "white",
    padding: "50px 25px 25px",
  },

  wrapperSlider: {
    position: "relative",
    outline: 0,
    "&:focus": {
      outline: "0 !important",
    },
    overflow: "auto",
    //maxHeight: "80%",
    backgroundColor: "white",
    //padding: "50px 25px 25px",
    width: "100%",
  },
}); */

const IModal = ({ open, type, onClose, children }: IModalProps) => {
  //const classes = useStyles();

  let wrapperClass = "";

  if (type === "form") wrapperClass = classes.wrapperForm;
  else if (type === "slider") wrapperClass = classes.wrapperSlider;

  return (
    <>
      {/* <Button variant="outlined" color="primary" onClick={handleToggle}>
        Show backdrop
      </Button> */}
      <Modal
        open={open}
        className={classes.modal}
        onClose={onClose}
        disableAutoFocus={true}
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
      >
        <div className={wrapperClass}>
          <ModalCloseButton
            onClick={onClose}
            ariaLabel="Закрыть модальное окно."
          />
          {children}
        </div>
      </Modal>
    </>
  );
};

export default IModal;
