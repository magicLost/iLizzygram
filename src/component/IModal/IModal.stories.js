import React from "react";
import { action } from "@storybook/addon-actions";
import IModal from ".";

export default {
  component: IModal,
  title: "Components/IModal",
  decorators: [],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

export const Default = () => {
  return (
    <>
      <IModal open={true} onClose={() => console.log("close")}>
        <div
          style={{
            width: "300px",
            height: "200px",
            textAlign: "center",
            backgroundColor: "white",
            padding: "20px",
          }}
        >
          <p>Hello.</p>
        </div>
      </IModal>
    </>
  );
};
