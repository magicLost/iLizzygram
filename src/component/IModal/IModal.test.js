import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import IModal from ".";
import classes from "./IModal.module.scss";

describe("IModal", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {
      _render = render(
        <IModal open={true} onClose={() => console.log("close")}>
          <div
            style={{
              width: "300px",
              height: "200px",
              textAlign: "center",
              backgroundColor: "white",
              padding: "20px",
            }}
          >
            <p>Hello.</p>
          </div>
        </IModal>
      );
    });

    afterEach(cleanup);

    describe("Snapshots", () => {
      test("matches snapshot", () => {
        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
