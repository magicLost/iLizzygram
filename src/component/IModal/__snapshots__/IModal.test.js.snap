// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IModal Render and props test Snapshots matches snapshot 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    aria-describedby="simple-modal-description"
    aria-labelledby="simple-modal-title"
    class="makeStyles-modal-1"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="makeStyles-wrapper-2"
      tabindex="-1"
    >
      <div
        class="makeStyles-root-3"
      >
        <button
          aria-label="Закрыть модальное окно."
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        style="width: 300px; height: 200px; text-align: center; background-color: white; padding: 20px;"
      >
        <p>
          Hello.
        </p>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
